drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,452 @@
+import functools
+import itertools
+import json
+import logging
+import multiprocessing
+import pathlib
+import shutil
+import subprocess
+import zipfile
+from typing import Any, BinaryIO, Dict, Iterator, List, Optional, Tuple, Union
+
+import capa
+import capa.capabilities
+import capa.capabilities.common
+import capa.engine
+import capa.features
+import capa.features.address
+import capa.features.common
+import capa.features.extractors
+import capa.features.extractors.base_extractor
+import capa.features.extractors.drakvuf
+import capa.features.extractors.drakvuf.extractor
+import capa.helpers
+import capa.loader
+import capa.main
+import capa.render
+import capa.render.json
+import capa.render.result_document
+import capa.rules
+import msgspec.json
+
+# rules related configuration
+capa_rules_dir = pathlib.Path("./capa-rules")
+capa_default_rules_repository_url = "https://github.com/mandiant/capa-rules/"
+
+# analysis configuration
+analyze_malware_pids_only = False
+perform_static_analysis = False
+perform_dynamic_analysis = True
+
+logger = logging.getLogger(__name__)
+
+
+def check_rules_exist(rules_dir: pathlib.Path) -> bool:
+    # this method checks whether there is a non-empty capa rules folder
+    try:
+        if any(rules_dir.iterdir()):
+            # return true if there exists a non-empty rules folder
+            return True
+        else:
+            # if the folder is empty, remove it and return false
+            shutil.rmtree(rules_dir)
+
+    except FileNotFoundError:
+        # if the rules folder does not exist, return false
+        pass
+
+    except NotADirectoryError:
+        # if there's a file with the same name as the rules' directory, delete it and return false
+        rules_dir.unlink()
+
+    # return false if no suitable capa rules folder was found
+    return False
```

Comment:
I understand why this code looks like it looks like (to support automatic git clone), but I think this function should just check for file existence (it can also get a more generic name):

```suggestion
def check_directory_exist(path: pathlib.Path) -> bool:
    # this method checks whether a non-empty directory exists
    return path.is_dir() and any(path.iterdir())
```